cpp/src/arrow/flight/sql/client.cc

```diff
@@ -256,6 +256,88 @@ arrow::Result<int64_t> FlightSqlClient::ExecuteSubstraitUpdate(
   return update_result.record_count();
 }
 
+arrow::Result<int64_t> FlightSqlClient::ExecuteIngest(
+      const FlightCallOptions& options,
+      const std::shared_ptr<RecordBatchReader>& reader,
+      const IngestMode& mode,
+      const std::string& table, const std::string& schema,
+      const std::string& catalog, const bool temporary,
+      const Transaction& transaction,
+      const std::map<std::string, std::string>& ingest_options) {
+  flight_sql_pb::CommandStatementIngest command;
+
+  flight_sql_pb::CommandStatementIngest_IngestMode pb_ingest_mode;
+  switch (mode)
+  {
```

Comment:
Make sure to run the linter/formatter

Reply:
I fixed some more linting issues that appear to be failing in CI. It's not clear to me that this resolves 100% of the issues because my local run of `archery lint --cpplint --clang-format --clang-tidy --fix` still exits with a nonzero status code but doesn't give a clear indication of what failed.

Two other CI checks appear to have failed as well, but it's not clear that they are related to these changes. Any insight into these failures on your end?